pkg/jobs/registry.go

```diff
@@ -368,6 +368,73 @@ func (r *Registry) newJob(record Record, jobID jobspb.JobID) *Job {
 	return job
 }
 
+// CreateJobsWithTxn creates jobs in a batch by including all the given job records
+// in a single insert to the jobs table.
+// There must be at least one job to create, otherwise the function returns an error.
+// If an error occurs while preparing the insert statement, no job is created.
+//TODO(sajjad): To discuss: What batch size do we expect? Should we limit the
```

Comment:
Yeah we'll want a maximum batch size. Start with just some constant. Maybe 100? We'll want to make sure we test using multiple batches.